File: api/src/main/java/api/Users/UserController.java
package api.Users;


import java.util.concurrent.atomic.AtomicLong;


import org.springframework.http.ResponseEntity;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;

@RestController
@RequestMapping("/user")
public class UserController{
  private final UserService USER_SERVICE;

  public UserController(UserService service){
   USER_SERVICE = service; 
  }
  // non-query actions
  @PostMapping("/register")
  public ResponseEntity<?> Register(@RequestBody RegisterDTO dto)
  { System.out.println("in function");  
    try{
      //create password logic 
      USER_SERVICE.Register(new UserDTO(dto.username, dto.email), dto.password);
      return new ResponseEntity<>(null, HttpStatus.OK);
    } catch(Exception ex) {
      return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
    }
  }



  @PatchMapping("/update")//probably will be in its own RestController AccountManagement
  public ResponseEntity<?> UpdateAccountUsername(@RequestBody String username,
                                                      @RequestBody String email)
  {
    try{
      
      USER_SERVICE.UpdateAccount(new UserDTO(username, email));
      return new ResponseEntity<>(null, HttpStatus.OK);
    } catch(Exception ex) {
      
      return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
    } 
  }



  @DeleteMapping("/delete")//probably will be in its own RestController AccountManagement
  public ResponseEntity<?> DeleteAccount()
  {
    try{
      
      USER_SERVICE.DeleteAccount("","");
      return new ResponseEntity<>(null, HttpStatus.OK);
    } catch(Exception ex) {
      
      return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
    } 
  }




  //query actions
  @GetMapping("/getinfo")//probably will be in its own RestController AccountManagement
  public ResponseEntity<String> GetAccountInfo(){ //prob have to change type 
    try{
      
      USER_SERVICE.GetAccountInfo(0L);//token a handle a travers spring security
      return new ResponseEntity<>(null, HttpStatus.OK);
    } catch(Exception ex) {
      
      return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
    } 
  }

}



class RegisterDTO {
    public String username;
    public String email;
    public String password;

}

File: api/src/main/java/api/controllers/UserController.kt
package api.controllers

import api.dtos.OptionalUserDto
import api.dtos.UserDTO
import api.dtos.UserLoginDTO
import api.dtos.UserRegisterationDto
import api.services.UserService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/user")
class UserController(
    private val service: UserService
) {
    @PostMapping("/register")
    fun register(
        @RequestBody registerationData: UserRegisterationDto
    ): ResponseEntity<UserDTO> {
        //create password logic
        val registeredUser = service.register(registerationData)
        return ResponseEntity<UserDTO>(registeredUser, HttpStatus.OK)
    }

    @PostMapping("/login")
    fun login(
        @RequestBody loginData: UserLoginDTO
    ): ResponseEntity<UserDTO> = service.login(loginData)
        .let { ResponseEntity<UserDTO>(it, HttpStatus.OK) }



    @PatchMapping("/{id}") // Probably will be in its own RestController AccountManagement
    fun updateUsername(
        @PathVariable id: Long,
        @RequestBody userUpdateData: OptionalUserDto
    ): ResponseEntity<UserDTO> {
        val updatedUser = service.update(id, userUpdateData)
        return ResponseEntity<UserDTO>(updatedUser, HttpStatus.OK)
    }


    @DeleteMapping("/{id}") // Probably will be in its own RestController AccountManagement
    fun deleteAccount(
        @PathVariable id: Long
    ): ResponseEntity<Void> {
        service.deleteAccount(id)
        return ResponseEntity<Void>(HttpStatus.OK)
    }


    //query actions
    @GetMapping("/{id}") // Probably will be in its own RestController AccountManagement
    fun getAccountInfo(
        @PathVariable id: Long
    ): ResponseEntity<UserDTO> {
        val dto = service.getById(id) // Token a handle a travers spring security
        return ResponseEntity<UserDTO>(dto, HttpStatus.OK)
    }
}





